### DISCUSSION TITLE
Network socket

### DISCUSSION THREAD

**Discussion by MalteKiefer** [netbox-community/netbox#10986] [Q&A]
Good morning,

I am just starting to use Netbox and I love it.
But now I come to a difficulty with a customer where I do not know exactly how to display it.

I have in one room a switch, under this switch are, instead of a patch panel two network sockets attached to the wall, once the supply line to the core switch and once to another switch in another room, see picture.
When I now connect the two switches in Netbox, currently all think these are connected directly and not via the network socket.

1. How can I know best represent the network outlet?
2. How can I then display the connection of the two switches above the network socket?

Thanks.

![image](https://user-images.githubusercontent.com/59220985/202889275-bce17423-6766-4098-808a-927b43026291.png)

**Comment by candlerb** [comment]
I would represent these wall sockets as separate Devices, each with one Front Port and one Rear Port.  Then the far-end room has another such device, and the Rear Ports are connected together with a cable.

Alternatively, you could represent all the wall sockets in a room as a single 'Device', and just create multiple front ports and rear ports on this device.

> once the supply line to the core switch

What's at the other end of this port?  If this is a "Circuit" from a telco provider, then you'd represent it as a "Circuit Termination" linked to a "Circuit".  You can then create a cable from this Circuit Termination to the core switch.

However, this is one area where Netbox's data model falls down though, in my opinion.  A "Circuit Termination" is a logical entity, not a physical port (e.g. it doesn't have a port type).  And yet to connect it, you create a Cable to it.  This means that one end of the cable is connected to a logical entity with an unspecified type of port, while the other end is connected to a 1000baseT Interface or whatever.

Furthermore, in reality, multiple provider "circuits" can be presented on a single physical port and cable.

**Comment by MalteKiefer** [comment]
My problem is, if I integrate the network socket, that I see then in the traceroute with the cables always only the next Job.
I don't have an overview where I can say I want to see how Switch A is connected to Switch B, with all the network sockets in between. 

![image](https://user-images.githubusercontent.com/59220985/202900086-e133c286-713d-4784-bc53-8cff0ab3c33b.png)

**Reply by candlerb** [reply]
Sorry, I don't understand what you mean.

When you create the socket, first you create a Rear Port, then you create a Front Port which is linked to that Rear Port. This tells Netbox it is a pass-through connection.  This gives you cable tracing _through_ the socket.

Therefore, if you connect the following three cables:

- switch A interface to socket A front port
- socket A rear port to socket B rear port
- socket B front port to switch B interface

then a cable trace from switch A interface will show you three segments, ending up at the switch B interface.  Try it!

Also, in the switch A Interfaces table, there are two columns you can enable:

* "Connection" shows the far end interface through the whole chain (i.e. switch B)
* "Link peers" shows the direct cable endpoint (i.e. socket A)

These columns are enabled with the "Configure Table" button.

If you want other graphical options, there is also Netbox Topology Views plugin.  Right now this only shows the cables+patch panels, but there is [work underway](https://github.com/mattieserver/netbox-topology-views/issues/151) to be able to show end-to-end connections too.
